test/gateway/eip20_gateway/stake.js

```diff
@@ -0,0 +1,324 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Test: stake.js
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const Gateway = artifacts.require("TestEIP20Gateway"),
+    MockToken = artifacts.require("MockToken"),
+    MessageBus = artifacts.require("MessageBus"),
+    GatewayLib = artifacts.require("GatewayLib");
+
+const utils = require("./../../test_lib/utils"),
+    BN = require('bn.js'),
+    EIP20GatewayKlass = require("./helpers/eip20_gateway"),
+    HelperKlass = require("./helpers/helper");
+
+const PENALTY_PERCENT = 1.5;
+
+let stakeAmount,
+    beneficiary,
+    stakerAddress,
+    gasPrice,
+    gasLimit,
+    nonce,
+    hashLock,
+    signature,
+    messageHash,
+    facilitator,
+    bountyAmount;
+
+let mockToken,
+    baseToken,
+    gateway,
+    helper,
+    hashLockObj,
+    gatewayTest,
+    errorMessage;
+
+
+async function _setup(accounts) {
+
+    mockToken = await MockToken.new();
+    baseToken = await MockToken.new();
+
+    bountyAmount = new BN(100);
+    gateway = await Gateway.new(
+        mockToken.address,
+        baseToken.address,
+        accounts[1], //core address
+        bountyAmount,
+        accounts[2] // organisation address
+    );
+
+    helper = new HelperKlass(gateway);
+    gatewayTest = new EIP20GatewayKlass(gateway, mockToken, baseToken);
+
+
+    hashLockObj = utils.generateHashLock();
+
+    facilitator = accounts[4];
+    nonce = await  helper.getNonce(accounts[1]);
+    stakeAmount = new BN(100000000000);
+    beneficiary = accounts[2];
+    stakerAddress = accounts[1];
```

Comment:
Same as organisation and core. On purpose?

Reply:
No, I didn't make any logic change in this file.  Though I have changed organisation and core addresses.